tests/DCPS/Compiler/vread_vwrite/VreadVwriteTest.cpp

```diff
@@ -276,11 +261,306 @@ TEST(VreadVwriteTest, SerializeTest)
   ASSERT_EQ(val["sa"][2], "south");
   ASSERT_EQ(val["sa"][3], "west");
+}
+
+TEST(VreadVwriteTest, StaticSerializeTest)
+{
+  Mod::Sample sample;
+  initialize_sample(sample);
+
+  // Serialize to JSON string.
+  Mod::SampleTypeSupport_var ts = new Mod::SampleTypeSupportImpl;
+  OpenDDS::DCPS::RepresentationFormat_var format = ts->make_format(OpenDDS::DCPS::JSON_DATA_REPRESENTATION);
+  CORBA::String_var buffer;
+  ASSERT_EQ(DDS::RETCODE_OK, ts->encode_to_string(sample, buffer, format));
+
+  DDS::OctetSeq_var bytes;
+  ASSERT_EQ(DDS::RETCODE_OK, ts->encode_to_bytes(sample, bytes, format));
+  const unsigned int n = bytes->length();
+  ASSERT_EQ(n, std::strlen(buffer));
+  ASSERT_EQ(0, std::memcmp(buffer.in(), bytes->get_buffer(), n));
+
+  // Then parse.
+  rapidjson::Document document;
+  document.Parse(buffer.in());
+  rapidjson::Value& val = document;
+  verify_parse_result(val);
 
   //rapidjson::StringBuffer buffer;
   //rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
   //val.Accept(writer);
   //std::string output(buffer.GetString());
   //std::cout << output << std::endl;
 }
+
+#ifndef OPENDDS_SAFETY_PROFILE
+
+DDS::DynamicType_var get_dynamic_type(OpenDDS::XTypes::TypeLookupService& tls)
+{
+  using namespace OpenDDS;
+  const XTypes::TypeIdentifier& ti = DCPS::getCompleteTypeIdentifier<DCPS::Mod_Sample_xtag>();
+  const XTypes::TypeMap& type_map = DCPS::getCompleteTypeMap<DCPS::Mod_Sample_xtag>();
+  const XTypes::TypeMap::const_iterator it = type_map.find(ti);
+  EXPECT_TRUE(it != type_map.end());
+
+  tls.add(type_map.begin(), type_map.end());
+  return tls.complete_to_dynamic(it->second.complete, DCPS::GUID_t());
+}
+
+TEST(VreadVwriteTest, DynamicSerializeTest)
+{
+  // Set up the dynamic data object
+  Mod::Sample sample;
+  initialize_sample(sample);
+
+  OpenDDS::XTypes::TypeLookupService tls;
+  DDS::DynamicType_var dt = get_dynamic_type(tls);
+  OpenDDS::XTypes::DynamicDataImpl dd(dt);
+  DDS::DynamicTypeMember_var dtm;
+  ASSERT_EQ(DDS::RETCODE_OK, dt->get_member_by_name(dtm, "id"));
+  ASSERT_EQ(DDS::RETCODE_OK, dd.set_int32_value(dtm->get_id(), sample.id));
+  ASSERT_EQ(DDS::RETCODE_OK, dt->get_member_by_name(dtm, "data"));
+  ASSERT_EQ(DDS::RETCODE_OK, dd.set_string_value(dtm->get_id(), sample.data));
+  ASSERT_EQ(DDS::RETCODE_OK, dt->get_member_by_name(dtm, "enu"));
+  ASSERT_EQ(DDS::RETCODE_OK, dd.set_int32_value(dtm->get_id(), sample.enu));
+  ASSERT_EQ(DDS::RETCODE_OK, dt->get_member_by_name(dtm, "enu2"));
+  ASSERT_EQ(DDS::RETCODE_OK, dd.set_int32_value(dtm->get_id(), sample.enu2));
+
+  // bt
+  ASSERT_EQ(DDS::RETCODE_OK, dt->get_member_by_name(dtm, "bt"));
+  DDS::DynamicData_var bt_dd;
+  ASSERT_EQ(DDS::RETCODE_OK, dd.get_complex_value(bt_dd, dtm->get_id()));
+  DDS::DynamicType_var bt_dt = bt_dd->type();
+
+  ASSERT_EQ(DDS::RETCODE_OK, bt_dt->get_member_by_name(dtm, "o"));
+  ASSERT_EQ(DDS::RETCODE_OK, bt_dd->set_byte_value(dtm->get_id(), sample.bt.o));
+#ifdef OPENDDS_HAS_EXPLICIT_INTS
```

Comment:
Does this still need to be conditional or can we assume that all builds have support for IDL explicit ints?

Reply:
It looks to me all the builds in CI have support for explicit ints. But it's configurable from TAO, so I think it still needs to be conditional here.

Reply:
OK, I guess removing the conditionals can be a separate PR